src/_pytest/config/__init__.py

```diff
@@ -408,7 +408,7 @@ def _getconftestmodules(self, path):
                 continue
             conftestpath = parent.join("conftest.py")
             if conftestpath.isfile():
-                mod = self._importconftest(conftestpath)
+                mod = self._importconftest(conftestpath.realpath())
```

Comment:
this needs a comment here after all it will cause the filename to be in disarray wrt the surrounding filenames

Reply:
Comment added